spec/queries/internal/moderators_query_spec.rb

```diff
@@ -0,0 +1,50 @@
+require "rails_helper"
+
+RSpec.describe Internal::ModeratorsQuery, type: :query do
+  let!(:user)  { create(:user, :trusted, name: "Greg") }
+  let!(:user2) { create(:user, :trusted, name: "Gregory") }
+  let!(:user3) { create(:user, :tag_moderator, name: "Paul", comments_count: 4) }
+  let!(:user4) { create(:user, :admin, name: "Susi", comments_count: 10) }
+  let!(:user5) { create(:user, :trusted, :admin, name: "Beth") }
+  let!(:user6) { create(:user, :admin, name: "Jean", comments_count: 5) }
```

Comment:
since this is a read only query, we don't probably need to recreate all these users at each test.
I think replacing `let!` with `let_it_be_readonly` is a good way to optimize in this case. 

More on that here:

https://github.com/thepracticaldev/dev.to/blob/9cebaa53e79c854ab9acc502de2363fa89073468/spec/support/initializers/test_prof.rb#L12